evol/population.py

```diff
@@ -142,9 +144,13 @@ def evolve(self, evolution: 'Evolution', n: int = 1) -> 'BasePopulation':  # noq
         :return: Population
```

Comment:
is there are reason why the conditions are not implemented such that they can be passed to the evolution object?

Reply:
TODO!

Reply:
After more consideration I think that we should provide **one** way to apply the conditions, as otherwise we may confuse users. We've discussed two options:

1. Contexts
```
with TimeLimit(seconds=300):
    result = pop.evolve(evo)
```

2. Argument
```
condition = TimeLimit(seconds=300)
result = pop.evolve(evo, condition=condition)
```

With multiple conditions these would become:
1. Contexts
```
with TimeLimit(seconds=300):
    with MinimumProgress(window=10):
        result = pop.evolve(evo)
```

2. Argument
```
conditions = [TimeLimit(seconds=300), MinimumProgress(window=10)]
result = pop.evolve(evo, condition=conditions)
```

I am in favour of option 1 (which is currently implemented), as:
-  it feels more explicit,
- the second requires the `condition` argument to accept both `Condition`s as well as lists of them, which I dislike.

@koaning what do you think?